services/catarse/catarse.js/legacy/src/c/project-row.js

```diff
@@ -20,9 +24,19 @@ const projectRow = {
                         ]),
                         m((showFriends ? '.w-col.w-col-4.w-col-small-6.w-col-tiny-6' : '.w-col.w-col-2.w-col-small-6.w-col-tiny-6'), [
                             m('.w-row', [
-                                (showFriends ? m('.w-col.w-col-6', [
-                                    m(`a.btn.btn-no-border.btn-small.btn-terciary[href="/connect-facebook?ref=${ref}"]`, 'Encontrar amigos')
-                                ]) : ''),
+                                (showFriends ? m('.w-col.w-col-6',
+                                    [
+                                        hasFBAuth ?
+                                        m(`a.btn.btn-no-border.btn-small.btn-terciary[href="/connect-facebook?ref=${ref}"]`, 'Encontrar amigos') :
+                                        m('form.button_to', {
+                                            action: `/users/auth/facebook`,
+                                            method: 'POST',
+                                        }, [
+                                            m('input.btn.btn-no-border.btn-small.btn-terciary.w-button[type=\'submit\'][value=\'Encontrar amigos\']'),
+                                            m(`input[name='authenticity_token'][type='hidden'][value='${h.authenticityToken()}']`),
+                                        ])
+                                    ]) : ''
```

Comment:
**Dúvida**
Vi que essa lógica está se repetindo basicamente por todo esse PR. Não seria possível criar um componente e reutilizá-lo aonde for necessário? Dessa forma tu mantém a lógica centralizada em um componente apenas, facilita uma manutenção futura.

Reply:
@stephannv eu também cheguei a pensar em criar um componente, mas embora a logica seja semelhante, as classes são diferentes e variam de acordo com a posição do link. E no caso do user-dropdown-profile-menu.tsx tem uma estrutura bem diferente das outras. Ainda assim, vale a pena fazer um componente?

Reply:
@gabrielras acho que vale. Veja alguns dos componentes `.tsx` que são definidos como função. As classes diferentes pode colocar como atributo e a estrutura diferente ai pode ser mais um componente ou cria um sub-componente que tenha a menor estrutura que se repete e um wrapper que tenha os parametros de classe ou estrutura.

Reply:
@grpse @stephannv show, vou ajustar a atividade para utilizar um componente.